db/commitlog/commitlog.cc

```diff
@@ -504,6 +504,16 @@ class db::commitlog::segment_manager : public ::enable_shared_from_this<segment_
     static inline constexpr auto immediately = message_clock::time_point::min();
     static inline constexpr auto no_timeout = message_clock::time_point::max();
 
+    struct write_request {
+        sseg_ptr segment = {};
+        buffer_type buffer = {};
+        uint64_t file_offset = 0;
+        size_t size = 0;
+        bool flush = false;
+        bool termination = false;
+    };
```

Comment:
Shouldn't writes have their on loop? Or even custom logic to avoid waiting to switch into the loop.

Certainly it can't be shared with slow ops.

Reply:
Write have their own queue/loop. Only terminating writes are handled by "slow" path, i.e. disposing of segments we no longer alloc into.